### PR TITLE
Reports per machine # 484

### PR DESCRIPTION
**Pull request by patriczek**
In this pull-request i made showing reports per machine (unique) on report page. The older view is still available after click on button "hide unique" - it's work as switch.
Also here was change the range of date, because for large range was text unreadable. The range is now set to 20 days, week and month.

### CODE REVIEW

**Comment by marusak** [comment]
Update please the alembic script. Yours down version does not match current head.

File: src/pyfaf/storage/migrations/versions/183a15e52a4f_report_history_unique.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Copyright (C) 2014  ABRT Team
```
Comment (mhabrnal) [inline src/pyfaf/storage/migrations/versions/183a15e52a4f_report_history_unique.py]: 2016

File: src/pyfaf/ureport.py [hunk 2]
```diff
@@ -276,8 +275,12 @@ def save_ureport2(db, ureport, create_component=False, timestamp=None, count=1):
         db_daily.opsysrelease = db_osrelease
         db_daily.day = day
         db_daily.count = 0
+        if "serial" in ureport["problem"] and ureport["problem"]["serial"] == 1:
```
Comment (jfilak) [inline src/pyfaf/ureport.py]: Why do you need this condition?

File: src/pyfaf/ureport.py [hunk 3]
```diff
@@ -288,8 +291,12 @@ def save_ureport2(db, ureport, create_component=False, timestamp=None, count=1):
         db_weekly.opsysrelease = db_osrelease
         db_weekly.week = week
         db_weekly.count = 0
+        if "serial" in ureport["problem"] and ureport["problem"]["serial"] == 1:
```
Comment (jfilak) [inline src/pyfaf/ureport.py]: Why do you need this condition?

File: src/pyfaf/ureport.py [hunk 4]
```diff
@@ -300,8 +307,12 @@ def save_ureport2(db, ureport, create_component=False, timestamp=None, count=1):
         db_monthly.opsysrelease = db_osrelease
         db_monthly.month = month
         db_monthly.count = 0
+        if "serial" in ureport["problem"] and ureport["problem"]["serial"] == 1:
```
Comment (jfilak) [inline src/pyfaf/ureport.py]: Why do you need this condition?

File: src/webfaf/templates/reports/item.html [hunk 5]
```diff
@@ -69,18 +69,6 @@
...
-            I know, now can be only one solution, but this is preparation
-            to future, when we want allow have a multiple solution
-          #}
-          <dt>Solution</dt>
-          <dd>
-              {% for solution in solutions %}
-                {{ solution.note_html }}<br>
-              {% endfor %}
-          </dd>
-          {% endif %}
```
Comment (jfilak) [inline src/webfaf/templates/reports/item.html]: Why did you remove this section?

File: src/webfaf/reports.py [hunk 6]
```diff
@@ -327,15 +325,87 @@ def item(report_id):
...
+                                  'opsysrelease_id': releases[0].ReportOpSysRelease.opsysrelease_id
+                                  })
+
+        if daily_history[0].day > (today - timedelta(MAX_DAYS)):
+            daily_history.append({'day': today - timedelta(MAX_DAYS),
+                                  'count': 0,
+                                  'opsysrelease_id': releases[0].ReportOpSysRelease.opsysrelease_id
+                                  })
+
+    # Show only 27 weeks
```
Comment (jfilak) [inline src/webfaf/reports.py]: 27?

File: src/webfaf/reports.py [hunk 7]
```diff
@@ -327,15 +325,87 @@ def item(report_id):
...
+            weekly_history.append({'week': last_monday,
+                                   'count': 0,
+                                   'opsysrelease_id': releases[0].ReportOpSysRelease.opsysrelease_id})
+
+        if weekly_history[0].week > ((last_monday - timedelta(7*MAX_WEEK)).date()):
+            weekly_history.append({'week': last_monday - timedelta(7*MAX_WEEK),
+                                   'count': 0,
+                                   'opsysrelease_id': releases[0].ReportOpSysRelease.opsysrelease_id})
+
+    # Show only 24 months
```
Comment (jfilak) [inline src/webfaf/reports.py]: 24?

File: src/webfaf/reports.py [hunk 8]
```diff
@@ -327,15 +325,87 @@ def item(report_id):
...
+                                                      .order_by(date)
+                                                      .all())
+
+    MAX_DAYS = 20  # Default set on 28
+    MAX_WEEK = 20  # Default set on 27
+    MAX_MONTH = 20  # Default set on 24
+
+    today = datetime.date.today()
+
+    # Show only 28 days
```
Comment (jfilak) [inline src/webfaf/reports.py]: 28?

File: src/webfaf/reports.py [hunk 9]
```diff
@@ -327,15 +325,87 @@ def item(report_id):
...
+    history_select = lambda table, date, date_range: (db.session.query(table).
+                                                      filter(table.report_id == report_id)
+                                                      .filter(date >= date_range)
+                                                      # Flot is confused if not ordered
+                                                      .order_by(date)
+                                                      .all())
+
+    MAX_DAYS = 20  # Default set on 28
+    MAX_WEEK = 20  # Default set on 27
+    MAX_MONTH = 20  # Default set on 24
```
Comment (jfilak) [inline src/webfaf/reports.py]: How are the comments about default related to the values?

File: src/webfaf/reports.py [hunk 10]
```diff
@@ -31,38 +48,19 @@
...
                    metric,
-                   metric_tuple,
                    request_wants_json,
                    is_component_maintainer)
 
-
 reports = Blueprint("reports", __name__)
 
-from webfaf_main import db, flask_cache, app
+from webfaf_main import db, flask_cache
```
Comment (jfilak) [inline src/webfaf/reports.py]: If you need to reorganize imports, please, do that in a solo commit.

File: src/webfaf/reports.py [hunk 11]
```diff
@@ -327,15 +325,87 @@ def item(report_id):
...
+    elif len(monthly_history) < MAX_MONTH:
+        if monthly_history[-1].month < (fdom):
+            monthly_history.append({'month': fdom,
+                                   'count': 0,
+                                   'opsysrelease_id': releases[0].ReportOpSysRelease.opsysrelease_id})
+
+        if monthly_history[0].month > (fdom - relativedelta(months=MAX_MONTH)):
+            monthly_history.append({'month': fdom - relativedelta(months=MAX_MONTH),
+                                   'count': 0,
+                                   'opsysrelease_id': releases[0].ReportOpSysRelease.opsysrelease_id})
```
Comment (jfilak) [inline src/webfaf/reports.py]: The change of history ranges should be done in a solo commit.

**Comment by sorki** [comment]
Thanks!
